src/server/routes/query/routes/visualization.ts

```diff
@@ -27,8 +30,53 @@ function getQuery(essence: Essence, timekeeper: Timekeeper): Expression {
   return essence.visualization.name === "grid" ? makeGridQuery(essence, timekeeper) : makeQuery(essence, timekeeper);
 }
 
+function start(clause: FixedTimeFilterClause): string {
+  return clause.values.first().start.toUTCString();
+}
+
+function intervalLength(clause: FixedTimeFilterClause): number {
+  const timeRange = clause.values.first();
+  return timeRange.end.getTime() - timeRange.start.getTime();
+}
+
+function timeVariables(essence: Essence, timekeeper: Timekeeper): Record<string, unknown> {
+  const timeFilter = essence.currentTimeFilter(timekeeper);
+  const timeDimension = essence.getTimeDimension();
+  const timeSplit = essence.splits.findSplitForDimension(timeDimension);
+
+  const startTime = start(timeFilter);
+  const interval = intervalLength(timeFilter);
+
+  const variables: Record<string, unknown> = { startTime, interval };
+
+  if (timeSplit && timeSplit.bucket instanceof Duration) {
+    variables.granularity = timeSplit.bucket.getDescription();
+  }
+
+  if (essence.hasComparison()) {
+    const previousTimeFilter = essence.previousTimeFilter(timekeeper);
+    variables.shiftedStartTime = start(previousTimeFilter);
+  }
+
+  return variables;
+}
+
+function logQueryInfo(essence: Essence, timekeeper: Timekeeper, logger: Logger) {
```

Comment:
I'm not sure that it should be a part of the route. It looks like a pure function and might be extracted and ... tested.

Reply:
It is a pure function in top level scope. If we'd like to test it we can do it without any issues.

Reply:
Agree, we could extract this function if needed in another route. As for now, only visualization queries are logged.